### PR TITLE
Refactor: Clean Up License

### PR DESCRIPTION
**Pull request by jens-erdmann**
Clean up license.py by removing debugging code and simplifying code.
See commits for details.

### CODE REVIEW

File: flict/flictlib/license.py [hunk 1]
```diff
@@ -340,91 +293,54 @@ def interim_license_expression_set_list(self, interim_license_expression_list):
...
-                    for lep_item in lep:
-                        logger.license_logger.debug(
-                            "    item: " + str(item) + " <--- " + str(lep_item))
-                        new_item = list(set(item + lep_item))
-                        logger.license_logger.debug("    item: " + str(item))
-                        new_list.append(new_item)
-                    logger.license_logger.debug("    list: " + str(new_list))
-                license_list = new_list
-            return license_list
+            raise Exception(f"Internal failure. Wrong type {lep} for: {lep}")
```
Comment (priv-kweihmann) [inline flict/flictlib/license.py]: Shouldn't that be a FlictError instead?
I honestly don't want to introduce plain Exceptions where it's not needed

File: flict/flictlib/license.py [hunk 2]
```diff
@@ -340,91 +293,54 @@ def interim_license_expression_set_list(self, interim_license_expression_list):
...
-                    "topsss: " + str(new_list) + " size: " + str(len(new_list)))
-            else:
-                new_list = license_list
-                logger.license_logger.debug("dapsss: " + str(license_list))
-                logger.license_logger.debug("dappss: " + str(lep))
-                for lep_item in lep:
-                    logger.license_logger.debug(
-                        "    item: " + str(license_list) + " <--- " + str(lep_item))
-                    new_list.append(lep_item)
+            raise Exception(f"Internal failure. Wrong type {lep} for: {lep}")
```
Comment (priv-kweihmann) [inline flict/flictlib/license.py]: Same here

File: flict/flictlib/license.py [hunk 3]
```diff
@@ -340,91 +293,54 @@ def interim_license_expression_set_list(self, interim_license_expression_list):
...
+
+        # single license
+        if len(license_list) == 0:
+            return lep
+
+        new_list = license_list
```
Comment (priv-kweihmann) [inline flict/flictlib/license.py:346]: I guess a `return list(lep)` could be used for this whole block here

File: flict/flictlib/license.py (hunk 1 above)
Comment (jens-erdmann) [reply flict/flictlib/license.py]: Good point. Was thinking about that myself when I crossed this.

Comment (jens-erdmann) [reply flict/flictlib/license.py]: Fixed

File: flict/flictlib/license.py (hunk 2 above)
Comment (jens-erdmann) [reply flict/flictlib/license.py]: Fixed

File: flict/flictlib/license.py (hunk 3 above)
Comment (jens-erdmann) [reply flict/flictlib/license.py:346]: But wouldn't that miss out on the potential content of `license_list`?

Comment (jens-erdmann) [reply flict/flictlib/license.py:346]: I gave it a spin and it fails a bunch of tests.

Comment (priv-kweihmann) [reply flict/flictlib/license.py:346]: True so it is `return list(license_list) + list(lep)` then - maybe also worth making it unique by doing `return list(set(list(license_list) + list(lep)))`

Comment (jens-erdmann) [reply flict/flictlib/license.py:346]: Just more errors.

Comment (priv-kweihmann) [reply flict/flictlib/license.py:346]: in this case - leave it for now as it is

**Review by priv-kweihmann — APPROVED** [review]

**Pull request merged**